Add version 16.0.1

mathieupotier opened the pull request:
Fixes : https://github.com/tuxgasy/docker-dolibarr/issues/101

New version of Dolibarr is out.
Create the image from PHP 8.1

Deprecated 7.3 images versions : this PHP is no more receiving security update.

The docker version now pass through build arg `DOLI_VERSION`.

mathieupotier commented on .gitignore:
```diff
@@ -1 +1,3 @@
+.idea/
+.volumes/
```
.idea/ is a dir created by my IDE : IntelliJ
.volumes/ is a directory used during the test / run defined in the docker-compose file.

it prevents me (and maybe others) to comit those files by mistakes.

mathieupotier commented on Dockerfile.template:
```diff
@@ -1,9 +1,10 @@
-ARG ARCH=
-
-FROM ${ARCH}php:%PHP_BASE_IMAGE%
```
I didn't see where it's used with buildx, and seems that it still worked with buildx.

Do you have a doc somewhere which refer to this arg ? for me as is, it's always blank "" and have no impact on removal.

tuxgasy replied:
It is required when building without buildx (see https://github.com/docker-library/official-images#architectures-other-than-amd64). It was my first attempt. Without, the image didn't work. But it was taking a long time and travis ci failed with timeout.

Maybe, with buildx, it's not required. Did you try an image built from x86-64 machine on an ARM machine ?

mathieupotier replied:
I'll try, I have a Pi4 available somewhere ...

mathieupotier commented on Dockerfile.template:
```diff
@@ -1,9 +1,10 @@
 
 LABEL maintainer="Garcia MICHEL <garcia@soamichel.fr>"
 
+ARG DOLI_VERSION
+ENV DOLI_VERSION ${DOLI_VERSION}
+
```
DOLI_VERSION is used on docker-run script, it needs this env var setup

tuxgasy replied:
Yes right. So what is the benefit of adding the ARG ?

mathieupotier replied:
I was preparing my other PR https://github.com/tuxgasy/docker-dolibarr/pull/103 and I estimate it was cleaner to get through build args, to pass args to build instead of having a template. but I admit, it's not point in this PR...

I'll do cleaning on code, separating the two subjects.
I always have more than one thing into my mind ... ^_^

mathieupotier commented on update.sh:
```diff
@@ -79,11 +65,14 @@ for dolibarrVersion in "${DOLIBARR_VERSIONS[@]}"; do
           "${dir}"
       else
-        docker build \
+        docker buildx build \
           --compress \
+          --platform linux/arm/v7,linux/arm64,linux/amd64 \
+          --build-arg DOLI_VERSION=${dolibarrVersion} \
           ${buildOptionTags} \
           "${dir}"
```
Yes but I wanted to exclude the "it works on my computer" effect ...
I'm preparing another branch to optimize few things (particularly the building process, with many more options), I can revert those changes in this PR for the moment.

mathieupotier replied:
I'm talking about those work in progress changes : https://github.com/tuxgasy/docker-dolibarr/pull/103 ^_^
Still having work on it, particularly on testing instance.

tuxgasy replied:
Ok. To be continued on the other PR.

mathieupotier commented:
reverted back change out of scope of this PR : just updated the version for Dolibarr new release, kept the image deprecation, and usage of PHP8.1 for 16.0.1, 16, latest and develop images.

tuxgasy commented:
It's good for me.

Maybe, we can switch also to bullseye (debian 11). What do you think ?

mathieupotier commented:
https://github.com/tuxgasy/docker-dolibarr/pull/103 <== here !!
I'll update the base image ... but yes, could be a good idea to use a more recent image.
